File: 429-n-ary-tree-level-order-traversal/429-n-ary-tree-level-order-traversal.py
from collections import deque

class Solution:
    def levelOrder(self, root: 'Node') -> List[List[int]]:
        # iterating approach with queue
        res = []
        if not root: return res
        queue = deque([root])
        
        while queue:
            size = len(queue)
            temp = []
            while size:
                node = queue.popleft()
                temp.append(node.val)
                for child in node.children:
                    queue.append(child)
                size -= 1
            res.append(temp)
        return res
        

File: 429-n-ary-tree-level-order-traversal/429-n-ary-tree-level-order-traversal.cpp
/*
// Definition for a Node.
class Node {
public:
    int val;
    vector<Node*> children;

    Node() {}

    Node(int _val) {
        val = _val;
    }

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
public:
    vector<vector<int>> levelOrder(Node* root) {
        if (!root) return {};
        queue<Node*> q;
        q.push(root);
        vector<vector<int>> res;
        
        while (!q.empty()) {
            int size = q.size();
            vector<int> temp;
            while (size) {
                Node* node = q.front(); q.pop();
                temp.push_back(node->val);
                for (auto child: node->children) {
                    q.push(child);
                }
                size--;
            }
            res.push_back(temp);
        }
        return res;
    }
};